ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +81,179 @@ - (void)testShouldIgnoreVnodeType {
+    
+    testVnode->values.getattr = FileFlags_IsInVirtualizationRoot;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertTrue(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = FileFlags_IsEmpty;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = FileFlags_Invalid;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->values.getattr = 100;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+}
+
+- (void)testShouldHandleVnodeOpEvent {
+    // In Parameters
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    testVnode->values.getattr = FileFlags_IsInVirtualizationRoot;
+    PerfTracer perfTracer;
+    vfs_context_t _Nonnull context = vfs_context_create(NULL);
+    kauth_action_t action = KAUTH_VNODE_READ_DATA;
+    
+    // Out Parameters
+    vtype vnodeType;
+    uint32_t vnodeFileFlags;
+    int pid;
+    char procname[MAXCOMLEN + 1] = "";
+    int kauthResult;
+    int kauthError;
+
+    
+    // Test Success Case
+    XCTAssertTrue(ShouldHandleVnodeOpEvent(
+                                            &perfTracer,
+                                            context,
+                                            testVnode.get(),
+                                            action,
+                                            &vnodeType,
+                                            &vnodeFileFlags,
+                                            &pid,
+                                            procname,
+                                            &kauthResult,
+                                            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+    
+    
+    // Test Invalid Action Bit
+    XCTAssertFalse(ShouldHandleVnodeOpEvent(
+                                           &perfTracer,
+                                           context,
+                                           testVnode.get(),
+                                           KAUTH_VNODE_ACCESS,
+                                           &vnodeType,
+                                           &vnodeFileFlags,
+                                           &pid,
+                                           procname,
+                                           &kauthResult,
+                                           &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+
+    
+    // Test invalid File System
```

Comment:
What's invalid about this file system?

Reply:
"hfs"

Reply:
That's [one of the 2 supported file systems](https://github.com/Microsoft/VFSForGit/blob/98acb8c7eaf92b26d514f68bf3f9dc88095ad475/ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp#L596) though?

FWIW, this test fails when I run it locally.

Reply:
Thanks for that, you are correct here.  Updating.